File: JavaScript_PlayGround/challenges/HackerRank/diagonal-difference.js
/*
 * Complete the 'diagonalDifference' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts 2D_INTEGER_ARRAY arr as parameter.
 */

/**
 * 
 * @param {[][]} arr
 * @return {number} 
 */
const  diagonalDifference = function(arr) {
    let totalL=0;
    let totalR = 0;
    let r =0;
    for(let m = 0 ; m < arr.length; m++ ){
        r = arr[m].length - 1 - m;
        totalL += arr[m][m];
        totalR += arr[m][r]; 
    }
    let diff =  totalL - totalR;
    return diff < 0 ? diff * -1 : diff;

}

let testCase=[
[1,2,3],
[4,5,6],
[9,8,9]
];
console.time();
console.log({value:testCase, result:diagonalDifference(testCase)});
console.timeEnd();

File: GO_PlayGround/console_go_application/challenges/hackerrank/diagonaldifference.go
package diagonaldifference

func Diagonaldifference(arr [3][3]int) int {
	var totalL int = 0
	var totalR int = 0
	var r int = 0
	for row := 0; row < len(arr); row++ {
		r = len(arr) - 1 - row
		totalL += arr[row][row]
		totalR += arr[row][r]
	}
	var diff int = totalL - totalR
	if diff < 0 {
		return diff * -1
	} else {
		return diff
	}

}
